__tests__/p2sh/tx-proposal-melt-tokens.test.js

```diff
@@ -0,0 +1,249 @@
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1,
+        change_address: changeAddress,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining(['wbe2eJdyZVimA7nJjmBQnKYJSXmpnpMKgG', changeAddress]));
+  });
+
+  it('should not accept melt token with empty change address', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1,
+        change_address: '',
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should not accept melt token with a change address that does not belong to the wallet', async () => {
+    const changeAddress = TestUtils.addresses[0];
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1,
+        change_address: changeAddress,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should return 200 with a valid body without melt authority', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1,
+        create_melt: false,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining(['wbe2eJdyZVimA7nJjmBQnKYJSXmpnpMKgG']));
+    expect(tx.outputs).toHaveLength(1);
+  });
+
+  it('should return 200 with a valid body selecting an authority address', async () => {
+    const testCases = [
+      // amount under 100 do not generate withdraw output
+      { amount: 1, expectedOutputLen: 2 },
+      // amount equal or greater than 100 generates withdraw output
+      { amount: 100, expectedOutputLen: 3 }
+    ];
+    testCases.forEach(cut => async () => { // cut: case under test
```

Comment:
Does this test finish correctly?

I feel that running an iterator of `async`s without `await`ing them on the test body would make them finish prematurely or leave them hanging failed after the test execution finishes.